symfony/plugins/orangehrmRecruitmentPlugin/Api/VacancyAPI.php

```diff
@@ -0,0 +1,387 @@
+    public const PARAMETER_STATUS = 'status';
+    public const PARAMETER_IS_PUBLISHED = 'isPublished';
+    public const PARAMETER_JOB_TITLE_ID = 'jobTitleId';
+    public const PARAMETER_EMPLOYEE_ID = 'employeeId';
+
+    public const PARAMETER_RULE_NAME_MAX_LENGTH = 100;
+    public const PARAMETER_RULE_JOB_TITLE_CODE_MAX_LENGTH = 4;
+    public const PARAMETER_RULE_HIRING_MANGER_ID_MAX_LENGTH = 13;
+    public const PARAMETER_RULE_NO_OF_POSITIONS_MAX_LENGTH = 13;
+    public const PARAMETER_RULE_STATUS_MAX_LENGTH = 4;
+    /**
+     * @var null|VacancyService
+     */
+    protected ?VacancyService $vacancyService = null;
+
+    /**
+     * @return EndpointResourceResult
+     * @throws DaoException
+     * @throws RecordNotFoundException
+     * @throws NormalizeException
+     */
+    public function getOne(): EndpointResourceResult
+    {
+        $id = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            CommonParams::PARAMETER_ID
+        );
+        $vacancy = $this->getVacancyService()->getVacancyById($id);
+        $this->throwRecordNotFoundExceptionIfNotExist($vacancy, Vacancy::class);
+        return new EndpointResourceResult(VacancyModel::class, $vacancy);
+    }
+
+    /**
+     * @return VacancyService
+     */
+    public function getVacancyService(): VacancyService
+    {
+        if (is_null($this->vacancyService)) {
+            $this->vacancyService = new VacancyService();
+        }
+        return $this->vacancyService;
+    }
+
+    /**
+     * @param  VacancyService  $vacancyService
+     */
+    public function setVacancyService(VacancyService $vacancyService)
+    {
+        $this->vacancyService = $vacancyService;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(new ParamRule(CommonParams::PARAMETER_ID));
+    }
+
+    /**
+     * @inheritDoc
+     * @throws DaoException
+     */
+    public function getAll(): EndpointCollectionResult
+    {
+        $vacancyParamHolder = new VacancySearchFilterParams();
+        $this->setSortingAndPaginationParams($vacancyParamHolder);
+
+        $vacancyParamHolder->setJobTitleId(
+            $this->getRequestParams()->getIntOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_JOB_TITLE_ID
+            )
+        );
+        $vacancyParamHolder->setEmpNumber(
+            $this->getRequestParams()->getIntOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_HIRING_MANAGER_ID
+            )
+        );
+        $vacancyParamHolder->setVacancyId(
+            $this->getRequestParams()->getIntOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_VACANCY_ID
+            )
+        );
+        $vacancyParamHolder->setStatus(
+            $this->getRequestParams()->getIntOrNull(
```

Comment:
`$this->getRequestParams()->getBooleanOrNull`

Reply:
In database status data type is integer. That's why I used int instead boolean